File: string_replacer.c
#include "stdio.h"
#include "stdlib.h"
#include "ctype.h"
#include "errno.h"
#include "string.h"


typedef enum{ERROR=-2, END=-1} error_returns;


int get_and_check_char(char *c){
	*c = getchar();
	if (!(isalpha(*c) || isspace(*c)))
		return EXIT_FAILURE;
	return EXIT_SUCCESS;
}

int is_end_of_sequence(char c){
	return c == '0';
}

int check_char(char c){
	if (!(isalpha(c)))
		return EXIT_FAILURE;
	return EXIT_SUCCESS;
}


int get_size_of_pattern_sequence(char *pattern, int i, char *c){
	int size = 1;
	if(*c == '0')
		return 0;
	while(*c != '0'){
		for(int index=0; index < i; index++){
			if(pattern[index] == *c){
				*c = getchar();
				if(check_char(*c) && !is_end_of_sequence(*c)){
					return ERROR;
				}
			}
			else{
				return size;
			}
		}
		++size;
	}
	return size;
}

void print_pattern(char* pattern, int i){
	for(int index = 0; index < i; index++){
		printf("%c", pattern[index]);
	}
}

int is_char_in_pattern(char *pattern, int i, char c){
	for(int index=0; index < i;index++){
		if(pattern[index] == c)
			return EXIT_FAILURE;
	}
	return EXIT_SUCCESS;
}

int adjust_size(int size){
	if(size > 9)
		return 0;
	return size;
}


int main()
{
	char c;
	int i = 0;
	int max = 9;
	int size = 0;
	char *pattern = (char *)malloc(10*sizeof(char));
	
	c = getchar();
	pattern[i] = c;
	++i;
	while(1){
		c = getchar();
		if(is_end_of_sequence(c))
			break;
		if(check_char(c)){
			fprintf(stderr, "ERROR: Wrong input. Contains non alphabetic characters\n");
			free(pattern);
			exit(EXIT_FAILURE);
		}
		if((i == 0) || (!is_char_in_pattern(pattern, i, c))){
			pattern[i] = c;
			++i;
			if (i == max){
				pattern = (char*)realloc(pattern, (10+max)*sizeof(char));
				max += 10;
			}
		}
		else{
			size = get_size_of_pattern_sequence(pattern, i, &c);
			if(size == ERROR){
				fprintf(stderr, "ERROR: Wrong input. Contains non alphabetic characters\n");
				free(pattern);
				exit(EXIT_FAILURE);
			}
			printf("%d", adjust_size(size));
			print_pattern(pattern, i);
			if(is_end_of_sequence(c))
				break;
			pattern[0]= c;
			i = 1;	
		}
	}
	size = get_size_of_pattern_sequence(pattern, i, &c);
	printf("1");
	print_pattern(pattern, i);
	free(pattern);
	return 0;
}

File: string_replacer.py
#!/usr/bin/env python
import sys


def is_end_of_listening(input_data):
    return input_data.startswith("-q")


def get_adjusted_output(sequence, cout):
    if cout > 9:
        return "0{0}".format(sequence)
    else:
        return "{0}{1}".format(cout, sequence)


def is_letter_in_sequence(letter, sequence):
    return sequence.find(letter) != -1


def get_count_and_erase_sequence(input_data, pattern):
    cout = 1
    while input_data.startswith(pattern):
        cout += 1
        input_data = input_data[len(pattern):]
    return input_data, cout


def replace_sequence(input_data):
    my_sequence = ""
    output = ""

    while input_data != "":
        if my_sequence == "" or not is_letter_in_sequence(input_data[0], my_sequence):
            my_sequence += input_data[0]
            input_data = input_data[1:]
        else:
            input_data, cout = get_count_and_erase_sequence(input_data, my_sequence)
            output += get_adjusted_output(my_sequence, cout)
            my_sequence = ""
    if my_sequence != "":
        output += get_adjusted_output(my_sequence, 1)
    print(output)

input_data = str(raw_input())
while not is_end_of_listening(input_data):
    if not input_data.isalpha():
        sys.exit("ERROR: Wrong input. Contains non alphabetic characters")
    replace_sequence(input_data)
    input_data = str(raw_input())
